rocketpy/mathutils/function.py

```diff
@@ -333,142 +345,173 @@ def set_extrapolation(self, method="constant"):
+        if interpolation == 0:  # linear
+
+            def linear_interpolation(x, x_min, x_max, x_data, y_data, coeffs):
+                x_interval = bisect_left(x_data, x)
+                x_left = x_data[x_interval - 1]
+                y_left = y_data[x_interval - 1]
+                dx = float(x_data[x_interval] - x_left)
+                dy = float(y_data[x_interval] - y_left)
+                y = (x - x_left) * (dy / dx) + y_left
+                return y
 
-        Returns
-        -------
-        self : Function
-        """
-        # Retrieve general info
-        x_data = self.x_array
-        y_data = self.y_array
-        x_min, x_max = self.x_initial, self.x_final
-        try:
-            extrapolation = self.EXTRAPOLATION_TYPES[self.__extrapolation__]
-        except KeyError as err:
-            raise ValueError(
-                f"Invalid extrapolation type '{self.__extrapolation__}'"
-            ) from err
+            self._interpolation_func = linear_interpolation
 
-        # Crete method to interpolate this info for each interpolation type
-        if self.__interpolation__ == "spline":
-            coeffs = self.__spline_coefficients__
+        elif interpolation == 1:  # polynomial
 
-            def get_value_opt(x):
-                if x_min <= x <= x_max:
-                    # Interpolate
-                    x_interval = np.searchsorted(x_data, x)
-                    x_interval = max(x_interval, 1)
-                    a = coeffs[:, x_interval - 1]
-                    x = x - x_data[x_interval - 1]
-                    y = a[3] * x**3 + a[2] * x**2 + a[1] * x + a[0]
-                else:
-                    # Extrapolate
-                    if extrapolation == 0:  # Extrapolation == zero
-                        y = 0
-                    elif extrapolation == 1:  # Extrapolation == natural
-                        if x < x_min:
-                            a = coeffs[:, 0]
-                            x = x - x_data[0]
-                        else:
-                            a = coeffs[:, -1]
-                            x = x - x_data[-2]
-                        y = a[3] * x**3 + a[2] * x**2 + a[1] * x + a[0]
-                    else:  # Extrapolation is set to constant
-                        y = y_data[0] if x < x_min else y_data[-1]
+            def polynomial_interpolation(x, x_min, x_max, x_data, y_data, coeffs):
+                y = np.sum(coeffs * x ** np.arange(len(coeffs)))
+                return y
+
+            self._interpolation_func = polynomial_interpolation
+
+        elif interpolation == 2:  # akima
+
+            def akima_interpolation(x, x_min, x_max, x_data, y_data, coeffs):
+                x_interval = bisect_left(x_data, x)
+                x_interval = x_interval if x_interval != 0 else 1
+                a = coeffs[4 * x_interval - 4 : 4 * x_interval]
+                y = a[3] * x**3 + a[2] * x**2 + a[1] * x + a[0]
                 return y
```

Comment:
Same as previous comment

```suggestion
                return a[3] * x**3 + a[2] * x**2 + a[1] * x + a[0]
```